### PR TITLE
Generalize `isUdaSelector` to handle any attribute

### PR DESCRIPTION
**Pull request by jacob-carlborg**
An upcoming PR will benefit from this.

### CODE REVIEW

**Comment by adamdruppe** [comment]
i like this model

File: src/dmd/objc.d [hunk 1]
```diff
@@ -733,11 +733,20 @@ extern(C++) private final class Supported : Objc
 
 extern(D) private:
 
-    bool isUdaSelector(StructDeclaration sd)
+    /**
+     * Returns `true` if the given symbol is a specific UDA declared in
+     * `core.attribute`.
```
Comment (MoonlightSentinel) [inline src/dmd/objc.d:738]:
```suggestion
     * `core.attribute` and has the idenfitier `ident`.
```

Comment (jacob-carlborg) [reply src/dmd/objc.d]: Done.

**Comment by jacob-carlborg** [comment]
All comments have been addressed.

**Review by thewilsonator — APPROVED** [review]

**Comment by jacob-carlborg** [comment]
Force pushed a new unchanged commit to retrigger the timed out test.

**Pull request merged**